### DISCUSSION TITLE
speed-up blockcheck

### DISCUSSION THREAD

**Discussion by NewUse** [bol-van/zapret#451] [Ideas]
RFC to parallelize/async mode of  blockcheck to speed-up the check process.
If I did understand correctly as for now blockeck use different modification of curl options to run curl and perform tests, but it is running requests modified by different statages one-by-one and is waiting for answer before sending new request. My idea is to make it parallel by either opening multiple sockets or using async mode, unfortunately it seems that the script should be rewritten in binary by using libcurl instead of curl binary file and a configuration file with same parameters that blockchek use. Libcurl also could be linked staticly and compiled with useful options such as quic support and doh support it could extend info about DPI and avoidence statages.

**Comment by bol-van** [comment]
dpi desync is performed by tpws or nfqws
in tpws case it's possible to use multiple socks instances on different ports
but nfqws/dvtws/winws are not connection bound
its possible to run multiple instances on different qnums/divert ports
and setup firewall rules for different curl uids
run curl with uid range
each to be redirected to its own pktws instance

binary coding is not required. its possible to code process pool in shell

this approach is not possible in windows
